### ISSUE TITLE
"HTTP session did not complete." vs "--directory-prefix" -> "No such file or directory"

### ISSUE THREAD

**Issue by SpiritQuaddicted** [chfoo/wpull#354]
`wpull --directory-prefix /tmp/nodirectoryexistshere www.time.com`

leads to

```
psutil: No module named 'psutil'. Resource monitoring will be unavailable.
INFO Fetching ‘http://www.time.com/’.
/home/me/.local/lib/python3.4/site-packages/wpull/protocol/http/client.py:185: UserWarning: HTTP session did not complete.
  warnings.warn(_('HTTP session did not complete.'))
ERROR FileNotFoundError: [Errno 2] No such file or directory: '/tmp/nodirectoryexistshere/tmp-wpull-resp_cb-d5qi9mtd.tmp'
INFO Exiting with status 3.
```

I notice this on other sites as well, for example www.google.com but not www.google.de.

```
$ ~/.local/bin/wpull3 --version
psutil: No module named 'psutil'. Resource monitoring will be unavailable.
2.0.1

$ python3 --version
Python 3.4.2

```

**Comment by JustAnotherArchivist** [comment]
In my opinion, there's no bug here. You tell wpull to write its files to an inexistent directory – of course that's going to fail. While wpull could try to create the directory with `os.makedirs`, I really don't think that's wpull's job.

The error could be improved though, namely by checking for the existence during the argument parsing already and erroring out immediately.

**Comment by SpiritQuaddicted** [comment]
wget does it ;)

Actually, wpull seems to do it now as well. But I still get the HTTP session warning for some reason.

Not sure if I use wpull or a fork though...

**Comment by JustAnotherArchivist** [comment]
You're right, it looks like `wpull.writer.BaseFileWriterSession.open_file` will create the directory if it doesn't exist. But as far as I can see, that has been the case since commit f3ebaa0d back in 2014 (first released in v0.4)...

The session warning might be completely unrelated. I've seen that warning quite a lot on ArchiveBot jobs before, and I have no idea what causes it.
